packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -209,14 +211,35 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
     /**
      * @internal
      */
-    public ariaLabelChanged(): void {
+    public ariaLabelChanged(_prev: unknown, _next: unknown): void {
         if (this.ariaLabel !== null && this.ariaLabel !== undefined) {
             this.editor.setAttribute('aria-label', this.ariaLabel);
         } else {
             this.editor.removeAttribute('aria-label');
         }
     }
 
+    /**
+     * @internal
+     */
+    public configurationChanged(
+        prev: EditorConfiguration | undefined,
+        next: EditorConfiguration
+    ): void {
+        if (this.isOnlyMentionInternalsChanged(prev, next)) {
+            this.setMarkdown(this.getMarkdown());
+        } else {
+            const currentStateMarkdown = this.getMarkdown();
+            this.richTextMarkdownSerializer = new RichTextMarkdownSerializer(
+                this.configuration?.mentionExtensionConfig.map(
+                    config => config.name
+                ) ?? []
```

Comment:
All the generics added to `Configuration<>` can be avoided and we should remove.
You just need a check here like:
```
this.configuration instanceof EditorConfiguration ? this.configuration.mentionExtensionConfig.map(config => config.name) : []
```

Reply:
Updated